File: SDKs/beezup.com/2.0/typescript/src/sdk/models/shared/updaterulerequest.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// UpdateRuleRequest
/** 
 * This update will replace the following values.
**/
export class UpdateRuleRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=endUtcDate" })
  endUtcDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=ruleName" })
  ruleName: string;

  @SpeakeasyMetadata({ data: "json, name=startUtcDate" })
  startUtcDate?: Date;
}

File: SDKs/beezup.com/2.0/python/src/sdk/models/shared/updaterulerequest.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class UpdateRuleRequest:
    r"""UpdateRuleRequest
    This update will replace the following values.
    """
    
    rule_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('ruleName') }})
    end_utc_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('endUtcDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    start_utc_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('startUtcDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    
